cfme/tests/intelligence/reports/test_reports.py

```diff
@@ -140,6 +140,29 @@ def rbac_api(appliance, request):
     )
 
 
+@pytest.fixture
+def restore_db(temp_appliance_preconfig, file_name):
+    db_file = FTPClientWrapper(cfme_data.ftpserver.entities.databases).get_file(file_name)
```

Comment:
This needs to be wrapped in a pytest.skip, and maybe some of the below calls in pytest.fail, to prevent this from resulting in an Error result when there's something amiss with the DB file fetching.

Reply:
**Question:** If something fails in the test setup fixture, should I use `pytest.skip` or` pytest.fail`?
When should I use `pytest.skip` and `pytest.fail`?

Reply:
Currently I've only used `pytest.skip` because I don't think I need to fail the test if it's setup failed. Let me know if you think otherwise.

Reply:
Up to your discretion, but I encourage behavior like:

If a data lookup fails, or something necessary for the test is missing, I would skip.

If an *action* during setup fails, I would use pytest.fail, because something we expect to work doesn't.  The core reason may be because something is missing, but there's always room for nuance.

Reply:
So I forgot to update the comment, but I used `pytest.fail` if the files fails to download on the appliance and `pytest.skip` if we fail to fetch the file from FTP server in the first step or if the downloaded file is broken.